eva-accession-release/src/main/java/uk/ac/ebi/eva/accession/release/io/AccessionedVariantMongoReader.java

```diff
@@ -95,8 +101,36 @@ public void open(ExecutionContext executionContext) throws ItemStreamException {
                                                                      submittedVariantValidated, allelesMatch,
                                                                      assemblyMatch, evidence);
 
-            addToVariants(variants, contig, start, rs, reference, alternate, sourceEntry);
+            addToVariants(variants, contig, submittedVariantStart, rs, reference, alternate, sourceEntry);
         }
         return new ArrayList<>(variants.values());
     }
+
+    /**
+     * The query performed in mongo can retrieve more variants than the actual ones because in some cases the same
+     * clustered variant is mapped against multiple locations. So we need to check that that clustered variant we are
+     * processing only appears in the VCF release file with the alleles from submitted variants matching the location.
+     */
+    private boolean isSameLocation(String contig, long start, String submittedVariantContig, long submittedVariantStart) {
+        return contig.equals(submittedVariantContig) && isSameStart(start, submittedVariantStart);
+    }
+
+    /**
+     * The start is considered to be the same when:
+     * - start in clustered and submitted variant match
+     * - start in clustered and submitted variant have a difference of 1
+     *
+     * The start position can be different in ambiguous INDELS because the renormalization is only applied to
+     * submitted variants. In those cases the start in the clustered and submitted variants will not exactly match but
+     * the difference should be 1
+     *
+     * Example:
+     * RS (assembly: GCA_000309985.1, accession: 268233057, chromosome: CM001642.1, start: 7356605, type: INS)
+     * SS (assembly: GCA_000309985.1, accession: 490570267, chromosome: CM001642.1, start: 7356604, reference: ,
+     *     alternate: AGAGCTATGATCTTCGGAAGGAGAAGGAGAAGGAAAAGATTCATGACGTCCAC)
+     */
+    private boolean isSameStart(long clusteredVariantStart, long submittedVariantStart) {
+         return clusteredVariantStart == submittedVariantStart
+                 || Math.abs(clusteredVariantStart - submittedVariantStart) == 1L;
```

Comment:
Just to be extra careful, can we also add a check for the variant type (INS or DEL) along with the difference check?

Reply:
Yes, I think that's a good idea. Check added